src/main/java/edu/ksu/canvas/attendance/controller/SummaryController.java

```diff
@@ -78,45 +82,80 @@ private ModelAndView studentSummary(String sectionId, String studentId, boolean
         if(student == null) {
             throw new IllegalArgumentException("Student does not exist in database.");
         }
+        student.getAttendances().sort(Comparator.comparing(Attendance::getDateOfClass).reversed());
 
         MakeupForm makeupForm = makeupService.createMakeupForm(validatedStudentId, validatedSectionId, addEmptyEntry);
 
         //Checking if Attendance Summary is Simple or Aviation
         AttendanceSection selectedSection = getSelectedSection(validatedSectionId);
         CourseConfigurationForm courseConfigurationForm = new CourseConfigurationForm();
-        boolean isSimpleAttendance = false;
-        if (selectedSection != null){
+        long selectedCourseId = 0;
+        if (selectedSection != null) {
             courseService.loadIntoForm(courseConfigurationForm, selectedSection.getCanvasCourseId());
-            isSimpleAttendance = courseConfigurationForm.getSimpleAttendance();
+            selectedCourseId = selectedSection.getCanvasCourseId();
         }
+        final boolean isSimpleAttendance = courseConfigurationForm.getSimpleAttendance();
 
         ModelAndView page = isSimpleAttendance ?
                 new ModelAndView("simpleStudentSummary") : new ModelAndView("studentSummary");
 
-        List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
-                reportService.getSimpleAttendanceSummaryReport(validatedSectionId) :
-                reportService.getAviationAttendanceSummaryReport(validatedSectionId);
-        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
+        List<AttendanceStudent> studentAttendanceList = studentService.getStudentByCourseAndSisId(student.getSisUserId(), selectedCourseId);
 
-        student.getAttendances().sort(Comparator.comparing(Attendance::getDateOfClass).reversed());
+        List<AttendanceSection> sectionList = new ArrayList<>();
+        for (int i = 0; i < studentAttendanceList.size(); i++) {
```

Comment:
maybe replace this loop with something like:
```
for(AttendanceStudent attendanceStudent: studentAttendanceList){
  AttendanceSection attendanceSection = sectionService.getSection(attendanceStudent.getCanvasSectionId();
  sectionList.add(attendanceSection);
}
```

this will remove the need for the .get(i).getCanvasSectionId() and improve the readability of the code, i think.